src/mscorlib/src/System/String.cs

```diff
@@ -2577,6 +2581,150 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
             }                        
         }
 
+#if FEATURE_CORECLR
+        [Pure]
+        private unsafe Boolean StartsWithOrdinal(String startsWith)
+        {
```

Comment:
Equals can do an int read to overrun the string length and include the null terminator. That might be hard to beat for short strings and make combining them difficult. Needs testing.
However the unrolling before that could be similar.